/* eslint-disable */
/* eslint new-cap: 0 */
/* global global_barista_token */   // from puptent
/* global global_barista_location */   // from puptent
/* global global_minerva_definition_name */   // from puptent
/* global model_id */   // from puptent
/* global global_model_type */   // from puptent

var jQuery = require('jquery');
var bbop = require('bbop').bbop;
var bbopx = require('bbopx');
var underscore = require('underscore');
var _ = underscore;
var graph_api = require('bbop-graph-noctua');
var minerva_requests = require('minerva-requests');
var angular = require('angular');
var widgetry = require('noctua-widgetry');
var selectize = require('selectize');

var annotationTitleKey = 'title';
var annotationSubjectKeyShorthand = 'dc11:subject';
var annotationSubjectKey = 'http://purl.org/dc/elements/1.1/subject';
var annotationTypeKey = 'http://purl.org/dc/elements/1.1/type';
var annotationTypeKeyShorthand = 'dc11:type';
var rootDiseaseId = 'DOID:4';  // https://monarchinitiative.org/disease/DOID:4
var rootPhenotypeId = 'UPHENO:0001001';  // https://monarchinitiative.org/phenotype/UPHENO:0001001
var rootOnsetId = 'HP:0003674';  // http://compbio.charite.de/hpoweb/showterm?id=HP:0003674
var rootEvidenceId = 'ECO:0000000';  // http://purl.obolibrary.org/obo/ECO_0000000

var USE_UI_GRID = false;

function createSolrAutocompleteForElement(element, options) {
  var jqElement = jQuery(element);
  jqElement.solrautocomplete(options);
}

var solrautocomplete = {
  createSolrAutocompleteForElement: createSolrAutocompleteForElement
};

var Solrautocomplete = window.Solrautocomplete = solrautocomplete;

jQuery.widget('widget.solrautocomplete', {
  options: {
    webserviceUrl: "http://localhost:8983/solr/select",
    onChange: console.log,
    required: false,
    optionDisplay: null,
    itemDisplay: null,
    valueField: null,
    searchField: null,
    queryData: null,
    golrManager: null,
    maxItems: 1
  },

  _create: function() {
    var that = this;
    // redefine the variables to avoid conflicts with selectize's scope
    var widgetOnChange = this.options.onChange;
    var widgetWebservicerUrl = this.options.webserviceUrl;
    var widgetRequired = this.options.required;
    var widgetOptionDisplay = this.options.optionDisplay;
    var widgetItemDisplay = this.options.itemDisplay;
    var widgetValueField = this.options.valueField;
    var widgetSearchField = this.options.searchField;
    var widgetFilterData = this.options.filterData;
    var widgetQueryData = this.options.queryData;
    var widgetGolrManager = this.options.golrManager;
    var widgetMaxItems = this.options.maxItems;
    var _currentValue = null;

    // Right label for user feedback
    var feedbackLabel = jQuery('<div class="feedback_label"></div>');
    this.element.before(feedbackLabel);

    var _checkSanity;
    var _updateHits;
    var _clearCache;

    var selectized = this.element.selectize({
      valueField: widgetValueField,
      searchField: widgetSearchField,
      create: false,
      render: {
        option: widgetOptionDisplay,
        item: widgetItemDisplay
      },
      load: function(query, callback) {
        if (!query.length) {
          return callback();
        }

        function customCallBack(res) {
          _updateHits(res._raw.response.numFound);
          callback(res._raw.response.docs);
        }
        widgetGolrManager.register('search', 'foo', customCallBack);
        if (widgetFilterData) {
          var filterData = widgetFilterData();
          _.each(filterData, function(f) {
            widgetGolrManager.add_query_filter(f[0], f[1]);
          });
        }

        if (widgetQueryData) {
          var queryData = widgetQueryData(query);
          widgetGolrManager.set_query(queryData);
        }

        widgetGolrManager.search();

        return null;
      },
      onChange: function(value) {
        widgetOnChange(value);
        _checkSanity(value, this);
        _currentValue = value;
      },
      onType: function(str) {
        if (widgetMaxItems === 1) {
          _clearCache(); // in order to request the server at each typing, only for single items
        }
      },
      onBlur: function() {
        _checkSanity(_currentValue);
      },
      maxItems: widgetMaxItems
    });

    this.selectized = selectized;

    _checkSanity = function(value) {
      if (widgetRequired && (value === null || value === '')) {
        feedbackLabel.show();
        feedbackLabel.text('Cannot be empty');
      }
      else {
        feedbackLabel.hide();
        feedbackLabel.text('');
      }
    };

    _clearCache = function() {
      selectized[0].selectize.clearCache("option");
      selectized[0].selectize.clearOptions();
    };

    _updateHits = function(h) {
      feedbackLabel.show();
      feedbackLabel.text(h + ' hits');
    };
  },

  _destroy: function() {
    this.selectized[0].selectize.destroy();
  },

  _setOptions: function(options) {
    this._super(options);
    // this.refresh();
  }
});


//
// Angular controller
//

function NoctuaBasicController($q, $scope, $animate, $timeout, $interval, $location, $anchorScroll, toastr, $window, $rootScope, uiGridConstants) {
  var that = this;

  this.$timeout = $timeout;
  this.$interval = $interval;
  this.$scope = $scope;
  this.$q = $q;
  this.USE_UI_GRID = USE_UI_GRID;

  if (USE_UI_GRID) {
    this.gridOptions = {
      rowHeight: 38,
      enableCellSelection: false,
      rowEditWaitInterval: -1,
      // enableCellEditOnFocus: false

      // http://embed.plnkr.co/RpNyqU/
      enableCellEdit: false, // set all columns to non-editable unless otherwise specified; cellEditableCondition won't override that
      enableCellEditOnFocus: true, // set any editable column to allow edit on focus
      cellEditableCondition: function(scope) {
        console.log('cellEditableCondition', scope.row.entity, that.editing_row);
        // put your enable-edit code here, using values from $scope.row.entity and/or $scope.col.colDef as you desire
        return scope.row.entity === that.editing_row; // in this example, we'll only allow active rows to be edited
      }
    };

    this.gridOptions.columnDefs = [
      // http://ui-grid.info/docs/#/tutorial
      // http://plnkr.co/edit/8ES5L4tPpA80u8x4tZWq?p=preview
      // http://plnkr.co/edit/uZjEV872L7FIUlFuZgHJ?p=preview
      {
        name: 'phenotype_label',
        field: 'phenotype',
        displayName: 'Phenotype',
        cellFilter: 'phenotypeLabel',
        sort: {
          direction: uiGridConstants.ASC,
          priority: 11
        },
        sortCellFiltered: true,
        width: '40%',
        editableCellTemplate: 'uiJQSelect',
        enableCellEdit: true
      },

      { name: 'ageofonset_label', displayName: 'Age of Onset' },
      { name: 'description', displayName: 'Description', enableCellEdit: true },
      { name: 'evidence_label', displayName: 'Evidence' },
      {
        name: 'actions',
        displayName: 'Actions',
        width: 100,
        resizable: false,
        cellTemplate: 'uigridActionCell',
        enableCellEdit: false,
        enableCellSelection: false,
        enableCellEditOnFocus: false,
        enableSorting: false
      }
    ];


    this.saveRowPromise = null;

    this.saveRow = function(rowEntity) {
      console.log('saveRow...this:', this, ' row:', rowEntity);
      that.selected_description = rowEntity.description;
      // // create a fake promise - normally you'd use the promise returned by $http or $resource
      var promise = this.$q.defer();
      this.saveRowPromise = promise;

      if (that.gridApi.rowEdit) {
        that.gridApi.rowEdit.setSavePromise(rowEntity, promise.promise);
      }

      that.$timeout(function() {
        promise.resolve();
      },
      100);
      // that.saveEditedRow();

      // var sortedData = that.gridApi.core.getVisibleRows();
      // console.log('xsortedData', sortedData.map(function(e) {
      //   return e.entity.phenotype;
      // }));
      // that.gridApi.cellNav.scrollToFocus(
      //   rowEntity,
      //   that.gridOptions.columnDefs[0]);
    };

    this.gridOptions.onRegisterApi = function(gridApi) {
      that.gridApi = gridApi;
      that.$scope.gridApi = gridApi;

      if (that.gridApi.rowEdit) {
        gridApi.rowEdit.on.saveRow(null, function(rowEntity) {
          console.log('rowEdit.on.saveRow', rowEntity);
          that.saveRow(rowEntity);
        });
      }

      gridApi.edit.on.beginCellEdit($scope, function(rowEntity, colDef) {
        console.log('beginCellEdit', rowEntity, colDef);
        var rowIndex = rowEntity.rowIndex;
        console.log('...rowIndex', rowIndex);

        if (!that.editing_row) {
          // that.$timeout(function() {
            that.editRow(rowEntity, rowIndex, false);
            if (colDef.field === 'phenotype') {
              that.editingPhenotype = true;
            }
          // }, 100);
        }
      });

      gridApi.edit.on.afterCellEdit($scope, function(rowEntity, colDef, newValue, oldValue) {
        console.log('afterCellEdit', oldValue, newValue);
        that.editingPhenotype = false;
        // that.saveRow(rowEntity);
      });

      that.$timeout(function() {
        that.gridApi.core.handleWindowResize();
        // var sortedData = that.gridApi.core.getVisibleRows();
        // console.log('sortedData', sortedData.map(function(e) {
        //   return e.entity.phenotype;
        // }));
        // that.gridApi.cellNav.scrollToFocus(
        //   sortedData[0],
        //   that.gridOptions.columnDefs[0]);
      });
    };
  }

  this.model_id = model_id;
  this.export_owl_url = '/download/' + this.model_id + '/owl';
  this.edit_graph_url = '/editor/graph/' + this.model_id;

  this.toastr = toastr;
  this.$window = $window;
  this.$rootScope = $rootScope;

  var phenotype_ageofonset_relation = "RO:0002488";
  var has_phenotype_relation = "BFO:0000051";

  // Events registry.
  var manager = null;
  var user_token = null;
  var graph = new graph_api.graph();
  this.grid_model = [];
  this.editing_row = null;
  this.new_row = null;
  this.selected_disease = null;
  this.selected_phenotype = null;
  this.selected_ageofonset = null;
  this.selected_evidence_metadata = [];
  this.selected_description = null;

  this.response_model = null;

  this.editingTitle = false;
  this.newTitle = null;
  this.modelTitle = null;

  this.editingSubject = false;
  this.modelSubject = null;
  this.modelSubjectMonarchKey = null;
  this.modelSubjectLabel = null;
  this.modelSubjectNodeId = null;
  this.modelType = global_model_type;

  //
  // Mapping between HPO Short Codes and either an ECO code or an HPO synthetic code.
  // Once the ECO incorporates HPO's codes, this code can be simplified or deleted.
  //
  var ecoPrefix = 'http://purl.obolibrary.org/obo/ECO_';
  var hpPrefix = 'http://human-phenotype-ontology.github.io/documentation.html#annot';
  this.HPShortCodes = {  /* eslint no-multi-spaces: 0 */
    IEA: {id: 'ECO:0000501',  label: 'Inferred from Electronic Annotation', url: ecoPrefix + '0000501'},
    ITM: {id: 'ECO:HPOITM',   label: 'Inferred by Text-mining', url: hpPrefix},
    TAS: {id: 'ECO:0000304',  label: 'Traceable Author Statement', url: ecoPrefix + '0000304'},
    PCS: {id: 'ECO:HPOPCS',   label: 'Published Clinical Study', url: hpPrefix},
    ICE: {id: 'ECO:HPOICE',   label: 'Individual Clinical Experience', url: hpPrefix}
  };
  this.HPECOToShortCode = {
    'ECO:0000501': 'IEA',
    'ECO:HPOITM': 'ITM',
    'ECO:0000304': 'TAS',
    'ECO:HPOPCS': 'PCS',
    'ECO:HPOICE': 'ICE'
  };


  this.newSubject = null;

  var compute_shield_modal = null;
  // Block interface from taking user input while
  // operating.
  function _shields_up() {
    if (compute_shield_modal) {
      // Already have one.
    }
    else {
      compute_shield_modal = bbopx.noctua.widgets.compute_shield();
      compute_shield_modal.show();
    }
  }

  // Release interface when transaction done.
  function _shields_down() {
    if (compute_shield_modal) {
      compute_shield_modal.destroy();
      compute_shield_modal = null;
    }
    else {
      // None to begin with.
    }
  }

  this.init = function() {
    // Only roll if the env is correct.

    // Try to define token.
    if (global_barista_token) {
      user_token = global_barista_token;
      that.user_token = user_token;

      if (user_token) {
        widgetry.user_check(global_barista_location,
          user_token, 'user_name_info');
        that.edit_graph_url = widgetry.build_token_link('/editor/graph/' + that.model_id, user_token);
      }
    }

    // Next we need a manager to try and pull in the model.
    if (typeof (global_minerva_definition_name) === 'undefined' ||
        typeof (global_barista_location) === 'undefined') {
      /* eslint no-alert: 0 */
      alert('environment not ready');
    }
    else {
      _shields_up();
      manager = new bbopx.minerva.manager(global_barista_location,
        global_minerva_definition_name,
        user_token);
      initializeAutocomplete();
      initializeCallbacks();
      manager.get_model(model_id);
    }
  };

  // type is 'error' or 'success'
  this.displayToast = function(type, msg) {
    var toastrFuncMap = {
      success: that.toastr.success,
      error: that.toastr.error
    };
    var toastrFunc = toastrFuncMap[type];
    toastrFunc(msg, msg);
  };


  function sanity_check() {
    if (that.selected_disease == "" || that.selected_disease == null) {
      that.displayToast('error', 'Disease cannot be empty.');
      return false;
    } else if (that.selected_phenotype == "" || that.selected_phenotype == null) {
      that.displayToast('error', 'Phenotype cannot be empty.');
      return false;
    } else {
      return true;
    }
  }

  this.store = function() {
    // this.setTitle(this.modelTitle);
    _shields_up();
    manager.store_model(model_id);
    _shields_down();
  };

  this.editTitle = function(optionalDefaultTitle) {
    that.newTitle = optionalDefaultTitle || that.modelTitle;
    that.editingTitle = true;
    $timeout(function () {
      angular.element('#modelTitle').focus();
    }, 10);
  };

  this.cancelEditTitle = function(newTitle) {
    that.editingTitle = false;
  };

  this.saveEditedTitle = function() {
    if (that.modelTitle !== that.newTitle) {
      _shields_up();
      var r = new minerva_requests.request_set(manager.user_token(), model_id);
      r.remove_annotation_from_model(annotationTitleKey, that.modelTitle);
      r.add_annotation_to_model(annotationTitleKey, that.newTitle);
      that.modelTitle = that.newTitle;
      r.store_model(model_id);
      manager.request_with(r, "edit_title");
      _shields_down();
    }
    that.editingTitle = false;
  };


  function isType(type, node) {
    var types = node._types;
    if (types.length == 1) {
      return types[0].class_id() === type;
    } else {
      return false;
    }
  }

  function getExistingIndividualId(id, nodes) {
    var hit = null;
    for (var key in nodes) {
      var value = nodes[key];
      if (isType(id, value)) {
        hit = value;
        break;
      }
    }
    if (hit === null) {
      return null;
    }
    else {
      return hit._id;
    }
  }

  this.saveEditedSubject = function() {
    if (that.modelSubject !== that.newSubject) {
      _shields_up();

      var r = new minerva_requests.request_set(manager.user_token(), model_id);
      r.remove_annotation_from_model(annotationSubjectKeyShorthand, that.modelSubject);
      r.remove_annotation_from_model(annotationTypeKey, that.modelType);
      r.remove_annotation_from_model(annotationTitleKey, that.modelTitle);
      r.add_annotation_to_model(annotationSubjectKey, that.newSubject);
      r.add_annotation_to_model(annotationTypeKey, that.modelType);
      r.add_annotation_to_model(annotationTitleKey, that.newTitle);

      var nodes = graph.get_nodes();

      var existing_disease_id = getExistingIndividualId(that.modelSubject, nodes);
      var subject_tmp_id = r.add_individual(that.newSubject);

      that.modelSubject = that.newSubject;
      that.modelSubjectMonarchKey = that.modelSubject.replace(/^MeSH:/, 'MESH:');
      that.modelTitle = that.newTitle;
      that.modelSubjectLabel = [that.newSubject, that.newSubject];
      that.modelSubjectNodeId = subject_tmp_id;
      that.editingSubject = true;
      manager.request_with(r, "edit_subject");
      _shields_down();
    }
  };



  function extract_class_id_from_node(node) {
    var types = node._types;
    if (types.length === 1) {
      return types[0].class_id();
    }
    else {
      return "";
    }
  }

  function extract_class_label_from_node(node) {
    var types = node._types;

    if (types.length === 1) {
      var result = types[0].class_label();
      if (that.modelType === 'hpo') {
        if (result.indexOf('ECO:HPO') === 0) {
          var code = result.slice(7);
          code = that.HPShortCodes[code];
          if (code) {
            result = code.label;
          }
        }
      }
      return result;
    }
    else {
      return "";
    }
  }

  function build_joined_label(id, label) {
    return id === label ? [label, label] : [id, label];
  }

  this.refresh_subject = function() {
    var annotations = graph.get_annotations_by_key(annotationSubjectKeyShorthand);
    if (annotations.length === 0) {
      // no subject set yet
    }
    else {
      var subject = annotations[0].value(); // there should be only one
      that.modelSubject = subject;
      that.modelSubjectMonarchKey = subject.replace(/^MeSH:/, 'MESH:');
      that.modelSubjectNodeId = subject;
      that.modelSubjectLabel = [subject, subject];
    }
  };

  this.refresh_title = function() {
    var annotations = graph.get_annotations_by_key(annotationTitleKey);
    if (annotations.length === 0) {
      // no title set yet
      // that.editTitle(that.modelSubjectLabel);
    }
    else {
      var title = annotations[0].value(); // there should be only one
      that.modelTitle = title;
    }
  };

  this.isValidModel = function () {
    return that.modelTitle && that.modelTitle.length > 0 &&
            that.grid_model.length > 0;
  };


  this.isValidAssociation = function () {
    return that.selected_disease &&
            that.selected_phenotype;
            // that.selected_ageofonset &&
            // that.selected_evidence_metadata.length > 0 &&
            // that.selected_evidence_metadata[0].ev &&
            // that.selected_evidence_metadata[0].ev.length &&
            // that.selected_description;
  };

  this.create = function() {
    var disease_id = that.selected_disease;
    var phenotype_id = that.selected_phenotype;
    var ageofonset_id = that.selected_ageofonset;
    var evidence_metadata = that.selected_evidence_metadata;
    var description = that.selected_description;

    _shields_up();
    if (sanity_check()) {
      var r = new minerva_requests.request_set(manager.user_token(), model_id);
      var nodes = graph.get_nodes();
      // fetching exising disease individual if it exists
      var existing_disease_id = getExistingIndividualId(disease_id, nodes);

      requestSetForCreation(r, disease_id, phenotype_id, ageofonset_id, evidence_metadata, description, existing_disease_id);
      var result = manager.request_with(r, "create");

    }
    else {
      _shields_down();
    }
  }


  this.addRow = function() {
    var row = {
      "disease": null,
      "disease_node_id": null,
      "disease_id": null,
      "disease_label": null,
      "phenotype": null,
      "phenotype_node_id": null,
      "phenotype_id": null,
      "phenotype_label": null,
      "ageofonset": null,
      "ageofonset_node_id": null,
      "ageofonset_id": null,
      "ageofonset_label": null,
      "evidence_metadata": [],
      "description": null
    };

    row.disease = this.modelSubject;
    row.disease_id = this.modelSubject;
    row.disease_node_id = this.modelSubjectNodeId;
    row.disease_label = this.modelSubjectLabel;

    if (that.USE_UI_GRID) {
      row.rowIndex = that.grid_model.length;
      row.phenotype = row.phenotype_label = row.phenotype_id = '';
      row.ageofonset = row.ageofonset_label = row.ageofonset_id = '';
    }
    that.grid_model.push(row);

    if (!that.USE_UI_GRID) {
      $timeout(function() {
        that.editRow(row, that.grid_model.length - 1, true);
      }, 10);
    }
  };


  this.editRow = function(row, rowIndex, isNew) {
    that.editing_row = row;
    that.new_row = isNew ? row : null;
    that.selected_disease = row.disease_id;
    that.selected_phenotype = row.phenotype_id;
    that.selected_ageofonset = row.ageofonset_id;
    var evidence_ids = underscore.map(row.evidence_metadata, function(ev) {
      return ev.id;
    });
    that.selected_evidence = evidence_ids;
    that.selected_description = row.description;
    that.selected_evidence_metadata = row.evidence_metadata;

    // keep them as reference for the deletion
    that.selected_disease_node_id_previous = row.disease_node_id;
    that.selected_disease_id_previous = row.disease_id;
    that.selected_phenotype_node_id_previous = row.phenotype_node_id;
    that.selected_ageofonset_node_id_previous = row.ageofonset_node_id;

    that.initializeRowAutocomplete(rowIndex);

    var phenotype_selectize = jQuery('#select_phenotype_' + rowIndex)[0];
    // console.log('phenotype_selectize', phenotype_selectize);
    if (phenotype_selectize) {
      phenotype_selectize = phenotype_selectize.selectize;
      phenotype_selectize.clearCache("option");
      phenotype_selectize.clearOptions();
      phenotype_selectize.addOption([{
        "id": row.phenotype_id,
        "annotation_class_label_searchable": row.phenotype_label
      }]);
      phenotype_selectize.setValue(row.phenotype_id);
    }

    var ageofonset_selectize = jQuery('#select_ageofonset_' + rowIndex)[0];
    if (ageofonset_selectize) {
      ageofonset_selectize = ageofonset_selectize.selectize;
      ageofonset_selectize.clearCache("option");
      ageofonset_selectize.clearOptions();
      ageofonset_selectize.addOption([{
        "id": row.ageofonset_id,
        "annotation_class_label_searchable": row.ageofonset_label
      }]);
      ageofonset_selectize.setValue(row.ageofonset_id);
    }

    that.ensure_ev(that.selected_evidence_metadata, rowIndex);
    var parentIndex = rowIndex;

    var evIndex = 0;
    underscore.each(that.selected_evidence_metadata, function(ev) {
      $timeout(function() {
        var ev_selector = '#' + ev.htmlid + '_' + parentIndex + '_' + evIndex;
        if (that.modelType !== 'hpo') {
          var ev_selectize = jQuery(ev_selector)[0];
          if (ev_selectize && ev_selectize.selectize) {
            ev_selectize = ev_selectize.selectize;
            // console.log('ev_selectize FOUND', ev_selector, ev_selectize);
          }
          else {
            Solrautocomplete.createSolrAutocompleteForElement(ev_selector, that.evidence_autocomplete_options(function(value) {
              $timeout(function() {
                ev.id = value;
              }, 10);
            }));

            ev_selectize = jQuery(ev_selector)[0].selectize;
            // console.log('ev_selectize CREATE', ev_selector, ev_selectize);
          }

          ev_selectize.clearCache("option");
          ev_selectize.clearOptions();
          ev_selectize.addOption([{
            "id": ev.id,
            "annotation_class_label_searchable": ev.label
          }]);
          ev_selectize.setValue(ev.id);
        }
        else {
          // xxx
        }

        ++evIndex;
      }, 10);
    });
    $timeout(function() {
      if (isNew) {
        $anchorScroll.yOffset = 200;
        $anchorScroll('scroll_anchor_' + rowIndex);
      }

      if (that.USE_UI_GRID) {
        that.gridApi.cellNav.scrollToFocus(
          row,
          that.gridOptions.columnDefs[0]);
      }

      phenotype_selectize = jQuery('#select_phenotype_' + rowIndex)[0];
      if (phenotype_selectize) {
        $timeout(function() {
          // console.log('phenotype_selectize.selectize', phenotype_selectize.selectize);
          phenotype_selectize.selectize.focus();
        }, 100);
      }
    }, 50);
  };


  this.cancelEditRow = function(rowIndex) {
    that.cleanupRowAutocomplete(rowIndex);

    if (that.new_row === underscore.last(that.grid_model)) {
      that.grid_model.pop();
    }

    that.editing_row = null;
  };




  function requestSetForDeletion(request_set, disease_node_id, phenotype_node_id, ageofonset_node_id) {
    var edges_from_disease = graph.get_edges_by_subject(disease_node_id);

    if (edges_from_disease.length === 1) { // delete only if there will be no edges left
      request_set.remove_individual(disease_node_id);
    }
    request_set.remove_individual(phenotype_node_id);
    if (ageofonset_node_id !== "") {
      request_set.remove_individual(ageofonset_node_id);
    }
  }

  function requestSetForCreation(request_set, disease_id, phenotype_id, ageofonset_id, evidence_metadata, description, existing_disease_id) {
    disease_id = disease_id.replace(/^MESH:/, 'MeSH:');

    var phenotype_tmp_id = request_set.add_individual(phenotype_id);
    if (existing_disease_id === null) {
      request_set.add_fact([request_set.add_individual(disease_id),
        phenotype_tmp_id,
        has_phenotype_relation
      ]);
    }
    else {
      request_set.add_fact([existing_disease_id,
        phenotype_tmp_id,
        has_phenotype_relation
      ]);
    }

    // Attach the metadata to the phenotype individual
    if (evidence_metadata !== "" &&
        evidence_metadata !== null &&
        evidence_metadata.length !== 0) {
      underscore.map(evidence_metadata, function(ev) {
        var evidence_tmp_id = request_set.add_individual(ev.id);
        request_set.add_annotation_to_individual("evidence", evidence_tmp_id, null, phenotype_tmp_id);
        var ref_list = ev.ref_list;
        if (ref_list !== "" && ref_list !== null && ref_list.length !== 0) {
          underscore.map(ref_list, function(ref) {
            // TODO create a proper individual when it'll supported by Minerva and Noctua
            // var ref_tmp_id = request_set.add_individual(ref.ref);
            // request_set.add_annotation_to_individual("source", ref_tmp_id, evidence_tmp_id);
            request_set.add_annotation_to_individual("source", ref.ref, null, evidence_tmp_id);
          });
        }
      });
    }

    // if (reference != "" && reference != null) {
    //   request_set.add_annotation_to_individual("source", reference, phenotype_tmp_id, model_id);
    // }

    if (description !== "" && description !== null) {
      request_set.add_annotation_to_individual("comment", description, null, phenotype_tmp_id);
    }

    if (ageofonset_id !== "" && ageofonset_id !== null) {
      request_set.add_fact([phenotype_tmp_id, request_set.add_individual(ageofonset_id), phenotype_ageofonset_relation]);
    }
  }

  this.saveEditedRow = function() {
    that.editing_row = null;

    _shields_up();

    var existing_disease = null;
    if (that.selected_disease_id_previous === that.selected_disease) {
      var edges_from_disease = graph.get_edges_by_subject(that.selected_disease_node_id_previous);
      if (edges_from_disease.length > 1) { // disease is not scheduled for deletion
        existing_disease = that.selected_disease_node_id_previous;
      }
    }

    var ev_ref_list = that.selected_evidence_metadata.map(function(e) {
      e = angular.copy(e);
      if (e.id in that.HPShortCodes) {
        e.id = 'HP:ECO_' + e.id;
      }
      return e;
    });

    if (ev_ref_list.length === 1 &&
        (!ev_ref_list[0].id || ev_ref_list[0].id === '')) {
      ev_ref_list = [];
    }

    // console.log('saveEditedRow...requestSetForDeletion:', that.selected_disease_node_id_previous, that.selected_phenotype_node_id_previous, that.selected_ageofonset_node_id_previous);
    // console.log('saveEditedRow...requestSetForCreation:', that.selected_disease, that.selected_phenotype, that.selected_ageofonset, ev_ref_list, that.selected_description, existing_disease);

    var r = new minerva_requests.request_set(manager.user_token(), model_id);
    requestSetForDeletion(r, that.selected_disease_node_id_previous, that.selected_phenotype_node_id_previous, that.selected_ageofonset_node_id_previous);
    requestSetForCreation(r, that.selected_disease, that.selected_phenotype, that.selected_ageofonset, ev_ref_list, that.selected_description, existing_disease);
    r.store_model(model_id);
    manager.request_with(r, "edit_row");
  };

  this.duplicateRow = function(sourceRow, sourceRowIndex) {
    var row = {
      "disease": sourceRow.disease,
      "disease_node_id": null,
      "disease_id": sourceRow.disease_id,
      "disease_label": sourceRow.disease_label,
      "phenotype": sourceRow.phenotype,
      "phenotype_node_id": null,
      "phenotype_id": sourceRow.phenotype_id,
      "phenotype_label": sourceRow.phenotype_label,
      "ageofonset": sourceRow.ageofonset,
      "ageofonset_node_id": null,
      "ageofonset_id": sourceRow.ageofonset_id,
      "ageofonset_label": sourceRow.ageofonset_label,
      "evidence_metadata": angular.copy(sourceRow.evidence_metadata),
      "description": sourceRow.description
    };
    row.rowIndex = that.grid_model.length;
    that.grid_model.push(row);
    $timeout(function() {
      that.editRow(row, row.rowIndex, true);
    }, 10);
  };

  this.deleteRow = function(disease_node_id, phenotype_node_id, ageofonset_node_id) {
    if (that.$window.confirm('Are you sure you want to delete this entry?')) {
      _shields_up();
      var r = new minerva_requests.request_set(manager.user_token(), model_id);
      requestSetForDeletion(r, disease_node_id, phenotype_node_id, ageofonset_node_id);
      manager.request_with(r, "remove_row");
    }
  };


  this.formatEVURL = function(ev) {
    var result = 'http://purl.obolibrary.org/obo/' + ev.id.replace(':', '_');

    if (this.modelType === 'hpo') {
      var hpoCode = this.HPECOToShortCode[ev.id];
      if (hpoCode) {
        result = this.HPShortCodes[hpoCode].url;
      }
    }

    return result;
  };


  this.formatEVLabel = function(ev) {
    var result = ev.id + ' ' + ev.label;

    if (this.modelType === 'hpo') {
      var hpoCode = this.HPECOToShortCode[ev.id];
      if (hpoCode) {
        result = hpoCode + ' ' + this.HPShortCodes[hpoCode].label;
      }
    }

    return result;
  };



  function activate_evidence_widget(ev, id, rowIndex, refIndex) {
    var ev_selector = '#' + ev.htmlid + '_' + rowIndex + '_' + refIndex;
    if (that.modelType !== 'hpo') {
      var ev_selectize = jQuery(ev_selector)[0];
      if (ev_selectize && ev_selectize.selectize) {
        ev_selectize = ev_selectize.selectize;
        // console.log('activate_evidence_widget FOUND', ev_selector, ev_selectize);
      }
      else {
        Solrautocomplete.createSolrAutocompleteForElement(ev_selector, that.evidence_autocomplete_options(function(value) {
          if (!ev) {
            console.log('activate_evidence_widget...undefined ev');
          }
          else {
            that.$timeout(function() {
              ev.id = value;
            }, 10);
          }
        }));
        ev_selectize = jQuery(ev_selector)[0];
        // console.log('activate_evidence_widget CREATE1', ev_selector, ev_selectize);
        ev_selectize = ev_selectize.selectize;
        // console.log('activate_evidence_widget CREATE2', ev_selector, ev_selectize);
      }

      ev_selectize.clearCache("option");
      ev_selectize.clearOptions();
    }
  }


  this.add_ev = function(selected_evidence_metadata, rowIndex) {
    var ev_id = 'ev' + selected_evidence_metadata.length;
    var ev = {
      'htmlid': ev_id
    };
    selected_evidence_metadata.push(ev);
    that.add_ref(ev);
    $timeout(function() {
      activate_evidence_widget(ev, ev_id, rowIndex, selected_evidence_metadata.length - 1);
    }, 10);
  };

  this.ensure_ev = function(selected_evidence_metadata, rowIndex) {
    if (selected_evidence_metadata.length === 0) {
      that.add_ev(selected_evidence_metadata, rowIndex);
    }
    else {
      underscore.each(that.selected_evidence_metadata, function(ev) {
        if (!ev.ref_list || ev.ref_list.length === 0) {
          that.add_ref(ev);
        }
      });
    }
  };

  this.lookup_ev_by_id = function(ev_id) {
    var result = null;
    underscore.each(that.selected_evidence_metadata, function(ev) {
      if (ev.id === ev_id) {
        result = ev;
      }
    });
    return result;
  };

  this.ensure_reflist = function(ev) {
    if (!ev.ref_list) {
      ev.ref_list = [];
    }
  };

  this.add_ref = function(ev) {
    this.ensure_reflist(ev);
    var ev_ref_id = ev.htmlid + 'ref' + ev.ref_list.length;
    var ev_ref = {
      'htmlid': ev_ref_id
    };
    ev.ref_list.push(ev_ref);
    // console.log('add_ref', ev, ev_ref);
  };

  function build_table() {
    var firstModelSubject = null;
    var firstModelSubjectLabel = '';

    that.grid_model = [];
    var edges = graph.all_edges();
    var has_phenotype_edges = underscore.filter(edges, function(edge) {
      return edge._predicate_id === has_phenotype_relation;
    });
    var phenotype_ageofonset_edges = underscore.filter(edges, function(edge) {
      return edge._predicate_id === phenotype_ageofonset_relation;
    });

    for (var i in has_phenotype_edges) {
      var current_edge = has_phenotype_edges[i];
      var edges_from_phenotype = graph.get_edges_by_subject(current_edge.target());
      var age_of_onset_node_id = "";
      var age_of_onset_id = "";
      var age_of_onset_label = "";
      var age_of_onset_display = "";
      if (edges_from_phenotype.length > 0) {
        var age_of_onset_node = graph.get_node(edges_from_phenotype[0].target());
        age_of_onset_id = extract_class_id_from_node(age_of_onset_node);
        age_of_onset_node_id = age_of_onset_node.id();
        age_of_onset_label = extract_class_label_from_node(age_of_onset_node);
        age_of_onset_display = build_joined_label(age_of_onset_id, age_of_onset_label);
      }

      var disease_node = graph.get_node(current_edge.source());
      var phenotype_node = graph.get_node(current_edge.target());
      var annotations = phenotype_node._annotations;

      var evidence_annotations = [];
      var description = "";
      for (var a in annotations) {
        var annotation = annotations[a];
        var key = annotation._properties.key;
        var value = annotation._properties.value;
        if (key === "evidence") {
          evidence_annotations.push(value);
        }
        else if (key === "comment") {
          description = value;
        }
      }

      var reference = [];
      /* eslint no-loop-func: 0 */
      // var evidences = underscore.map(evidence_annotations, function(ev) {
      //   var evidence_node = graph.get_node(ev);
      //   return build_joined_label(extract_class_id_from_node(evidence_node), extract_class_label_from_node(evidence_node));
      // });
      var counter = 0;
      var evidence_metadata = underscore.map(evidence_annotations, function(ev_node_id) {
        counter += 1;
        var evidence_node = graph.get_node(ev_node_id);
        var evidence_node_annotations = evidence_node._annotations;
        var current_evidence_refs = [];

        // TODO bad reference display string at the moment
        for (var e in evidence_node_annotations) {
          var eannotation = evidence_node_annotations[e];
          var ekey = eannotation._properties.key;
          var evalue = eannotation._properties.value;
          if (ekey === "source") {
            reference.push(evalue);
            current_evidence_refs.push({
              "ref": evalue,
              "htmlid": "ref" + (e + 1)
            });
          }
        }

        var ev = {
          node_id: evidence_node._id,
          id: extract_class_id_from_node(evidence_node),
          xid: extract_class_id_from_node(evidence_node),
          label: extract_class_label_from_node(evidence_node),
          ref_list: current_evidence_refs,
          htmlid: "ev" + counter
        };

        return ev;
      });

      // if (evidences.length === 0) {
      //   evidences = "";
      // }

      var entry = {
        "disease_node_id": disease_node.id(),
        "disease_id": extract_class_id_from_node(disease_node),
        "disease_label": extract_class_label_from_node(disease_node),
        "phenotype_node_id": phenotype_node.id(),
        "phenotype_id": extract_class_id_from_node(phenotype_node),
        "phenotype_label": extract_class_label_from_node(phenotype_node),
        "ageofonset_node_id": age_of_onset_node_id,
        "ageofonset_id": age_of_onset_id,
        "ageofonset_label": age_of_onset_label,
        "evidence_metadata": evidence_metadata,
        "description": description
      };

      entry.index = entry.disease_id + '/' + entry.phenotype_id + '/' + age_of_onset_id;
      entry.disease = build_joined_label(entry.disease_id, entry.disease_label);
      entry.phenotype = build_joined_label(entry.phenotype_id, entry.phenotype_label);
      entry.ageofonset = build_joined_label(entry.ageofonset_id, entry.ageofonset_label);

      if (that.USE_UI_GRID) {
        entry.rowIndex = that.grid_model.length;
      }

      that.grid_model.push(entry);
    }

    if (that.USE_UI_GRID) {
      that.gridOptions.data = that.grid_model;
    }
  }


  function refresh_ui() {
    build_table();
    that.refresh_subject();
    that.refresh_title();
    that.$timeout(function () {
      that.initializeSubjectAutocomplete();
    }, 100);
    // if (that.user_token) {
    //   that.ensure_ev(that.selected_evidence_metadata);
    // }
  }

  function manager_changed(resp, man) {
    var data = resp.data();
    that.response_model = JSON.stringify(resp);

    var tmp_graph = new graph_api.graph();
    tmp_graph.load_data_basic(data);
    graph = tmp_graph;
    // console.log('manager_changed', graph);
    if (that.saveRowPromise) {
      that.saveRowPromise.resolve();
      that.saveRowPromise = null;
    }
    if (that.editingSubject) {
      var individuals = data.individuals;
      var subjectId = individuals[0].type[0].id;

      // console.log('subjectId:', subjectId);
      // console.log('that.modelTitle:', that.modelTitle);
      // console.log('that.newTitle:', that.newTitle);
      that.modelSubjectLabel = [subjectId, subjectId];
      that.editingSubject = false;
    }
    else {
      that.selected_disease = null;
      that.selected_phenotype = null;
      that.selected_ageofonset = null;

      that.selected_evidence_metadata = [];
      that.selected_description = null;

      that.$rootScope.$apply(refresh_ui);

      _shields_down();
    }
  }

  function initializeCallbacks() {
    manager.register('manager_error', 'manager_errorx', function(resp, man) {
      console.log("manager_error");
      console.log(resp);
      console.log(man);
      _shields_down();
      that.displayToast("error", resp._message);
    }, 10);

    manager.register('error', 'errorargh', function(resp, man) {
      console.log("error");
      console.log(resp);
      _shields_down();
      that.displayToast("error", resp._message);
    }, 10);

    manager.register('rebuild', 'foorebuild', manager_changed, 10);
    manager.register('merge', 'merdge', manager_changed, 10);
  }

  function initializeAutocomplete() {
    //
    // The first golr_json_string below is from the original WebPhenote (2015) version.
    // The second golr_json_string is from the monarch-app/conf/golr-conf.json
    // TODO read this golr_json conf from a file copied (at build time or periodically) from Monarch App or
    // some more central repo.
    //
    var golr_json_string = '{"generic_association":{"result_weights":"subject^7.0 subject_category^6.2 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0","filter_weights":"subject_category^8.0 subject_taxon^7.5 subject_closure_label^7.0 relation_closure_label^6.5 evidence_closure_label^6.0 object_category^5.0 object_closure_label^4.5","_infile":"/Users/cjm/repos/monarch-app/conf/golr-views/oban-config.yaml","display_name":"Annotations","description":"Associations following OBAN model","schema_generating":"true","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields":[{"id":"id","type":"uuid","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","property":[]},{"id":"subject","type":"string","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Subject","property":[]},{"searchable":"true","id":"subject_label","type":"string","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Subject","property":[]},{"cardinality":"multi","id":"subject_closure","type":"string","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Subject (Inferred)","property":[]},{"searchable":"true","cardinality":"multi","id":"subject_closure_label","type":"string","description":"Labels for subject_closure.","display_name":"Subject (Inferred)","property":[]},{"id":"subject_category","type":"string","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","property":[]},{"id":"subject_category_label","type":"string","description":"Label for association_subject_category","display_name":"Subject category","property":[]},{"searchable":"true","id":"subject_description","type":"string","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","property":[]},{"cardinality":"multi","id":"subject_synonym","type":"string","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","property":[]},{"id":"subject_taxon","type":"string","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Taxon","property":[]},{"searchable":"true","id":"subject_taxon_label","type":"string","description":"Label of taxon.","display_name":"Taxon","property":[]},{"cardinality":"multi","id":"subject_taxon_closure","type":"string","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","property":[]},{"searchable":"true","cardinality":"multi","id":"subject_taxon_closure_label","type":"string","description":"Labels for taxon_closure.","display_name":"Taxon","property":[]},{"searchable":"true","id":"subject_family","type":"string","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","property":[]},{"searchable":"true","id":"subject_family_label","type":"string","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","property":[]},{"id":"relation","type":"string","description":"A relationship type that connects the subject with object. TODO: check for correspondence in OBAN","display_name":"Relationship","property":[]},{"id":"relation_label","type":"string","description":"Label for association_relation","display_name":"Relationship","property":[]},{"cardinality":"multi","id":"relation_closure","type":"string","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Inferred relationship","property":[]},{"cardinality":"multi","id":"relation_closure_label","type":"string","description":"labels for association_relation_closure","display_name":"Inferred relationship","property":[]},{"cardinality":"multi","id":"qualifier","type":"string","description":"Association qualifier. TBD: NEGATION should probably be handled differently somehow","display_name":"Qualifier","property":[]},{"id":"object","type":"string","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Object","property":[]},{"searchable":"true","id":"object_label","type":"string","description":"Label for association_object.","display_name":"Object","property":[]},{"cardinality":"multi","id":"object_closure","type":"string","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Object (Inferred)","property":[]},{"searchable":"true","cardinality":"multi","id":"object_closure_label","type":"string","description":"Labels for object_class_closure.","display_name":"Object (Inferred)","property":[]},{"id":"object_category","type":"string","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","property":[]},{"id":"object_category_label","type":"string","description":"Label for association_object_category","display_name":"Object type","property":[]},{"id":"object_taxon","type":"string","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Taxon","property":[]},{"searchable":"true","id":"object_taxon_label","type":"string","description":"Label of taxon.","display_name":"Taxon","property":[]},{"cardinality":"multi","id":"object_taxon_closure","type":"string","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","property":[]},{"searchable":"true","cardinality":"multi","id":"object_taxon_closure_label","type":"string","description":"Labels for taxon_closure.","display_name":"Taxon","property":[]},{"id":"object_isoform","type":"string","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","property":[]},{"cardinality":"multi","id":"object_class_closure","type":"string","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","property":[]},{"searchable":"tru","cardinality":"multi","id":"object_class_closure_label","type":"string","description":"Labels for object_class_closure.","display_name":"Involved in","property":[]},{"cardinality":"multi","id":"object_class_secondary_closure","type":"string","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","property":[]},{"searchable":"true","cardinality":"multi","id":"object_class_secondary_closure_label","type":"string","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","property":[]},{"cardinality":"multi","id":"object_extension","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},{"searchable":"true","cardinality":"multi","id":"object_extension_label","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},{"cardinality":"multi","id":"object_extension_closure","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},{"searchable":"true","cardinality":"multi","id":"object_extension_closure_label","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},{"cardinality":"multi","id":"object_extension_json","type":"string","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","property":[]},{"id":"source","type":"string","description":"TBD: ALIGN GO AND DC TERMINOLOGY. Source of association. The association may be derived from multiple source, but this should be the outer source, if applicable","display_name":"Source","property":[]},{"id":"date","type":"string","description":"Date of assignment. The association may be derived from multiple source, but this should be the outer source, if applicable.","display_name":"Date","property":[]},{"id":"assigned_by","type":"string","description":"TBD: ALIGN GO AND DC TERMINOLOGY. Source of association. The association may be derived from multiple source, but this should be the outer source, if applicable","display_name":"Assigned by","property":[]},{"id":"is_redundant_for","type":"string","description":"Rational for redundancy of annotation.","display_name":"Redundant for","property":[]},{"id":"evidence_graph","type":"string","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","property":[]},{"cardinality":"multi","id":"evidence","type":"string","description":"Evidence type. In Monarch we may have a graph/chain. This is always the outer evidence OR a list of all evidence types used?","display_name":"Evidence type","property":[]},{"cardinality":"multi","id":"evidence_label","type":"string","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","property":[]},{"cardinality":"multi","id":"evidence_closure","type":"string","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","property":[]},{"cardinality":"multi","id":"evidence_closure_label","type":"string","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence label closure","property":[]},{"cardinality":"multi","id":"evidence_object","type":"string","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence from","property":[]},{"cardinality":"multi","id":"citation","type":"string","description":"citation.","display_name":"Reference","property":[]}],"fields_hash":{"object_taxon_closure":{"cardinality":"multi","id":"object_taxon_closure","type":"string","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","property":[]},"object_class_secondary_closure_label":{"searchable":"true","cardinality":"multi","id":"object_class_secondary_closure_label","type":"string","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","property":[]},"date":{"id":"date","type":"string","description":"Date of assignment. The association may be derived from multiple source, but this should be the outer source, if applicable.","display_name":"Date","property":[]},"subject_family_label":{"searchable":"true","id":"subject_family_label","type":"string","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","property":[]},"object_isoform":{"id":"object_isoform","type":"string","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","property":[]},"object_taxon":{"id":"object_taxon","type":"string","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Taxon","property":[]},"relation_label":{"id":"relation_label","type":"string","description":"Label for association_relation","display_name":"Relationship","property":[]},"object_taxon_closure_label":{"searchable":"true","cardinality":"multi","id":"object_taxon_closure_label","type":"string","description":"Labels for taxon_closure.","display_name":"Taxon","property":[]},"subject_closure":{"cardinality":"multi","id":"subject_closure","type":"string","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Subject (Inferred)","property":[]},"subject_taxon_label":{"searchable":"true","id":"subject_taxon_label","type":"string","description":"Label of taxon.","display_name":"Taxon","property":[]},"id":{"id":"id","type":"uuid","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","property":[]},"subject_closure_label":{"searchable":"true","cardinality":"multi","id":"subject_closure_label","type":"string","description":"Labels for subject_closure.","display_name":"Subject (Inferred)","property":[]},"subject_taxon_closure":{"cardinality":"multi","id":"subject_taxon_closure","type":"string","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","property":[]},"citation":{"cardinality":"multi","id":"citation","type":"string","description":"citation.","display_name":"Reference","property":[]},"subject":{"id":"subject","type":"string","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Subject","property":[]},"subject_label":{"searchable":"true","id":"subject_label","type":"string","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Subject","property":[]},"subject_category":{"id":"subject_category","type":"string","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","property":[]},"evidence_closure":{"cardinality":"multi","id":"evidence_closure","type":"string","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","property":[]},"subject_taxon":{"id":"subject_taxon","type":"string","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Taxon","property":[]},"relation":{"id":"relation","type":"string","description":"A relationship type that connects the subject with object. TODO: check for correspondence in OBAN","display_name":"Relationship","property":[]},"evidence_object":{"cardinality":"multi","id":"evidence_object","type":"string","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence from","property":[]},"assigned_by":{"id":"assigned_by","type":"string","description":"TBD: ALIGN GO AND DC TERMINOLOGY. Source of association. The association may be derived from multiple source, but this should be the outer source, if applicable","display_name":"Assigned by","property":[]},"object_closure_label":{"searchable":"true","cardinality":"multi","id":"object_closure_label","type":"string","description":"Labels for object_class_closure.","display_name":"Object (Inferred)","property":[]},"evidence_label":{"cardinality":"multi","id":"evidence_label","type":"string","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","property":[]},"source":{"id":"source","type":"string","description":"TBD: ALIGN GO AND DC TERMINOLOGY. Source of association. The association may be derived from multiple source, but this should be the outer source, if applicable","display_name":"Source","property":[]},"object_extension_closure_label":{"searchable":"true","cardinality":"multi","id":"object_extension_closure_label","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},"object":{"id":"object","type":"string","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Object","property":[]},"object_extension_label":{"searchable":"true","cardinality":"multi","id":"object_extension_label","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},"object_extension_closure":{"cardinality":"multi","id":"object_extension_closure","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},"subject_family":{"searchable":"true","id":"subject_family","type":"string","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","property":[]},"relation_closure":{"cardinality":"multi","id":"relation_closure","type":"string","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Inferred relationship","property":[]},"object_extension_json":{"cardinality":"multi","id":"object_extension_json","type":"string","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","property":[]},"subject_taxon_closure_label":{"searchable":"true","cardinality":"multi","id":"subject_taxon_closure_label","type":"string","description":"Labels for taxon_closure.","display_name":"Taxon","property":[]},"qualifier":{"cardinality":"multi","id":"qualifier","type":"string","description":"Association qualifier. TBD: NEGATION should probably be handled differently somehow","display_name":"Qualifier","property":[]},"object_extension":{"cardinality":"multi","id":"object_extension","type":"string","description":"Extension class for the annotation.","display_name":"Annotation extension","property":[]},"evidence":{"cardinality":"multi","id":"evidence","type":"string","description":"Evidence type. In Monarch we may have a graph/chain. This is always the outer evidence OR a list of all evidence types used?","display_name":"Evidence type","property":[]},"object_label":{"searchable":"true","id":"object_label","type":"string","description":"Label for association_object.","display_name":"Object","property":[]},"object_closure":{"cardinality":"multi","id":"object_closure","type":"string","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Object (Inferred)","property":[]},"subject_synonym":{"cardinality":"multi","id":"subject_synonym","type":"string","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","property":[]},"object_class_secondary_closure":{"cardinality":"multi","id":"object_class_secondary_closure","type":"string","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","property":[]},"object_class_closure":{"cardinality":"multi","id":"object_class_closure","type":"string","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","property":[]},"subject_category_label":{"id":"subject_category_label","type":"string","description":"Label for association_subject_category","display_name":"Subject category","property":[]},"relation_closure_label":{"cardinality":"multi","id":"relation_closure_label","type":"string","description":"labels for association_relation_closure","display_name":"Inferred relationship","property":[]},"evidence_graph":{"id":"evidence_graph","type":"string","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","property":[]},"object_class_closure_label":{"searchable":"tru","cardinality":"multi","id":"object_class_closure_label","type":"string","description":"Labels for object_class_closure.","display_name":"Involved in","property":[]},"object_taxon_label":{"searchable":"true","id":"object_taxon_label","type":"string","description":"Label of taxon.","display_name":"Taxon","property":[]},"object_category_label":{"id":"object_category_label","type":"string","description":"Label for association_object_category","display_name":"Object type","property":[]},"subject_description":{"searchable":"true","id":"subject_description","type":"string","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","property":[]},"is_redundant_for":{"id":"is_redundant_for","type":"string","description":"Rational for redundancy of annotation.","display_name":"Redundant for","property":[]},"evidence_closure_label":{"cardinality":"multi","id":"evidence_closure_label","type":"string","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence label closure","property":[]},"object_category":{"id":"object_category","type":"string","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","property":[]}},"document_category":"generic_association","weight":"20","_strict":0,"id":"generic_association","_outfile":"/Users/cjm/repos/monarch-app/conf/golr-views/oban-config.yaml"}}';
    golr_json_string = '{"disease_pathway":{"id":"disease_pathway","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^8.0 relation^7.0 subject^6.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Disease","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Pathway","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Pathway","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Pathway","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Pathway","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Pathway Taxon","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Pathway Taxon","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Disease","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Disease","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Pathway","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Pathway","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Pathway","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Pathway","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Pathway Taxon","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Pathway Taxon","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"disease_phenotype":{"id":"disease_phenotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Disease","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Disease","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Disease","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"dovechart":{"id":"dovechart","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_category^6.2 subject_taxon^6.5 relation^6.0  object^5.0 object_category^4.5 object_taxon^4.2 evidence^4.0 source^3.5","filter_weights":"subject_taxon^7.5 object_taxon^4.7","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Subject","type":"string","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Subject","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Object","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Object","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Object Taxon","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Object Taxon","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence from","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Subject","type":"string","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Subject","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Object","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Object","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Object Taxon","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Object Taxon","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence from","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"gene_disease":{"id":"gene_disease","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"gene_homolog":{"id":"gene_homolog","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^7.0 object_taxon^6.5 relation^6.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Homolog","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Homolog","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Homolog","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Homolog","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Homolog Taxon","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Homolog Taxon","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Homolog Taxon","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Homolog Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Homolog","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Homolog","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Homolog","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Homolog","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Homolog Taxon","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Homolog Taxon","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Homolog Taxon","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Homolog Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"gene_pathway":{"id":"gene_pathway","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^8.0 relation^7.0 subject^6.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Pathway","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Pathway","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Pathway","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Pathway","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Pathway Taxon","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Pathway Taxon","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Pathway","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Pathway","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Pathway","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Pathway","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Pathway Taxon","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Pathway Taxon","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Pathway Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"gene_phenotype":{"id":"gene_phenotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Gene","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"genotype_gene_on_gene_page":{"id":"genotype_gene_on_gene_page","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Genotype","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Genotype","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"genotype_disease":{"id":"genotype_disease","schema_Diseaserating":true,"document_category":"Diseaseric_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, Disease, person, ....","display_name":"Genotype","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For Diseases will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a Disease-Disease interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"schema_generating":true,"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, Disease, person, ....","display_name":"Genotype","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For Diseases will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a Disease-Disease interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"genotype_gene":{"id":"genotype_gene","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 object_taxon^4.2 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Genotype","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Genotype","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"genotype_phenotype":{"id":"genotype_phenotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Genotype","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Genotype","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Genotype","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"literature_disease":{"id":"literature_disease","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_label^6.5 is_defined_by^3.0","filter_weights":"object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"literature_gene":{"id":"literature_gene","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^6.5 is_defined_by^3.0","filter_weights":"object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"literature_genotype":{"id":"literature_genotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^6.5 is_defined_by^3.0","filter_weights":"object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Genotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Genotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Genotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"literature_model":{"id":"literature_model","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^6.5 is_defined_by^3.0","filter_weights":"object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Model","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Model","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Model","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Model","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"literature_phenotype":{"id":"literature_phenotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^6.5 is_defined_by^3.0","filter_weights":"object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"literature_variant":{"id":"literature_variant","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"object^6.5 is_defined_by^3.0","filter_weights":"object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Variant","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Variant","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Publication ID","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Title","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Publication","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Publication","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Variant","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Variant","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"model_disease":{"id":"model_disease","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"model_gene":{"id":"model_gene","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 object_taxon^4.2 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"model_genotype":{"id":"model_genotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"relation_closure_label^6.5 evidence_closure_label^6.0 subject_closure_label^4.5 object_closure_label^4.0 subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Genotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Genotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Genotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"model_phenotype":{"id":"model_phenotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"model_variant":{"id":"model_variant","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 object_taxon^4.2 object_gene^4.1 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Variant","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Variant","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Model","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Model","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Model","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Model","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Variant","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Variant","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"generic_association":{"id":"generic_association","schema_generating":true,"description":"Associations following OBAN model","display_name":"Annotations","document_category":"generic_association","weight":20,"boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","result_weights":"subject^7.0 subject_category^6.2 subject_taxon^6.5 relation^6.0  object^5.0 object_category^4.5 object_taxon^4.2 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^7.5 object_taxon_label^4.7","fields":[{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Subject","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Subject","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Object","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Object","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Object Taxon","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Object Taxon","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence from","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"fields_hash":{"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Subject","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Subject","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Subject (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Subject Taxon","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Subject Taxon","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Taxon","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Object","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Object","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Object Taxon","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Object Taxon","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Object Taxon","type":"string","cardinality":"multi","property":[],"searchable":true},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence from","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"ontology":{"id":"ontology","schema_generating":true,"description":"Gene Ontology Term, Synonym, or Definition.","display_name":"Ontology","document_category":"ontology_class","weight":40,"boost_weights":"annotation_class^3.0 annotation_class_label^5.5 description^1.0 comment^0.5 synonym^1.0 alternate_id^1.0 regulates_closure^1.0 regulates_closure_label^1.0","result_weights":"annotation_class^8.0 description^6.0 source^4.0 synonym^3.0 alternate_id^2.0","filter_weights":"source^4.0 subset^3.0 regulates_closure_label^1.0 is_obsolete^0.0","fields":[{"id":"id","description":"Term identifier.","display_name":"Acc","type":"string","property":["getIdentifier"]},{"id":"annotation_class","description":"Term identifier.","display_name":"Term","type":"string","property":["getIdentifier"]},{"id":"annotation_class_label","description":"Identifier.","display_name":"Term","type":"string","property":["getLabel"],"searchable":true},{"id":"description","description":"Term definition.","display_name":"Definition","type":"string","property":["getDef"],"searchable":true},{"id":"source","description":"Term namespace.","display_name":"Ontology source","type":"string","property":["getNamespace"]},{"id":"is_obsolete","description":"Is the term obsolete?","display_name":"Obsoletion","type":"boolean","property":["getIsObsoleteBinaryString"]},{"id":"comment","description":"Term comment.","display_name":"Comment","type":"string","property":["getComment"],"searchable":true},{"id":"synonym","description":"Term synonyms.","display_name":"Synonyms","type":"string","property":["getOBOSynonymStrings"],"cardinality":"multi","searchable":true},{"id":"alternate_id","description":"Alternate term identifier.","display_name":"Alt ID","type":"string","property":["getAnnotationPropertyValues","alt_id"],"cardinality":"multi"},{"id":"replaced_by","description":"Term that replaces this term.","display_name":"Replaced By","type":"string","property":["getAnnotationPropertyValues","replaced_by"],"cardinality":"multi"},{"id":"consider","description":"Others terms you might want to look at.","display_name":"Consider","type":"string","property":["getAnnotationPropertyValues","consider"],"cardinality":"multi"},{"id":"subset","description":"Special use collections of terms.","display_name":"Subset","type":"string","property":["getSubsets"],"cardinality":"multi"},{"id":"definition_xref","description":"Definition cross-reference.","display_name":"Def xref","type":"string","property":["getDefXref"],"cardinality":"multi"},{"id":"database_xref","description":"Database cross-reference.","display_name":"DB xref","type":"string","property":["getXref"],"cardinality":"multi"},{"id":"isa_partof_closure","description":"Ancestral terms (is_a/part_of).","display_name":"Is-a/part-of","type":"string","cardinality":"multi","property":["getRelationIDClosure","BFO:0000050"]},{"id":"isa_partof_closure_label","description":"Ancestral terms (is_a/part_of).","display_name":"Is-a/part-of","type":"string","cardinality":"multi","property":["getRelationLabelClosure","BFO:0000050"],"searchable":true},{"id":"regulates_closure","description":"Ancestral terms (regulates, occurs in, capable_of).","display_name":"Ancestor","type":"string","cardinality":"multi","property":["getRelationIDClosure","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"]},{"id":"regulates_closure_label","description":"Ancestral terms (regulates, occurs in, capable_of).","display_name":"Ancestor","type":"string","cardinality":"multi","property":["getRelationLabelClosure","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"],"searchable":true},{"id":"topology_graph_json","description":"JSON blob form of the local stepwise topology graph. Uses various relations (including regulates, occurs in, capable_of).","display_name":"Topology graph (JSON)","type":"string","property":["getSegmentShuntGraphJSON","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"],"indexed":false},{"id":"regulates_transitivity_graph_json","description":"JSON blob form of the local relation transitivity graph. Uses various relations (including regulates, occurs in, capable_of).","display_name":"Regulates transitivity graph (JSON)","type":"string","property":["getLineageShuntGraphJSON","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"],"indexed":false},{"id":"only_in_taxon","description":"Only in taxon.","display_name":"Only in taxon","type":"string","property":["getIdentifier"],"searchable":true},{"id":"only_in_taxon_label","description":"Only in taxon label.","display_name":"Only in taxon","type":"string","property":["getLabel"],"searchable":true},{"id":"only_in_taxon_closure","description":"Only in taxon closure.","display_name":"Only in taxon (IDs)","type":"string","cardinality":"multi","property":["getRelationLabelClosure","RO:0002160"]},{"id":"only_in_taxon_closure_label","description":"Only in taxon label closure.","display_name":"Only in taxon","type":"string","cardinality":"multi","property":["getRelationLabelClosure","RO:0002160"],"searchable":true}],"fields_hash":{"id":{"id":"id","description":"Term identifier.","display_name":"Acc","type":"string","property":["getIdentifier"]},"annotation_class":{"id":"annotation_class","description":"Term identifier.","display_name":"Term","type":"string","property":["getIdentifier"]},"annotation_class_label":{"id":"annotation_class_label","description":"Identifier.","display_name":"Term","type":"string","property":["getLabel"],"searchable":true},"description":{"id":"description","description":"Term definition.","display_name":"Definition","type":"string","property":["getDef"],"searchable":true},"source":{"id":"source","description":"Term namespace.","display_name":"Ontology source","type":"string","property":["getNamespace"]},"is_obsolete":{"id":"is_obsolete","description":"Is the term obsolete?","display_name":"Obsoletion","type":"boolean","property":["getIsObsoleteBinaryString"]},"comment":{"id":"comment","description":"Term comment.","display_name":"Comment","type":"string","property":["getComment"],"searchable":true},"synonym":{"id":"synonym","description":"Term synonyms.","display_name":"Synonyms","type":"string","property":["getOBOSynonymStrings"],"cardinality":"multi","searchable":true},"alternate_id":{"id":"alternate_id","description":"Alternate term identifier.","display_name":"Alt ID","type":"string","property":["getAnnotationPropertyValues","alt_id"],"cardinality":"multi"},"replaced_by":{"id":"replaced_by","description":"Term that replaces this term.","display_name":"Replaced By","type":"string","property":["getAnnotationPropertyValues","replaced_by"],"cardinality":"multi"},"consider":{"id":"consider","description":"Others terms you might want to look at.","display_name":"Consider","type":"string","property":["getAnnotationPropertyValues","consider"],"cardinality":"multi"},"subset":{"id":"subset","description":"Special use collections of terms.","display_name":"Subset","type":"string","property":["getSubsets"],"cardinality":"multi"},"definition_xref":{"id":"definition_xref","description":"Definition cross-reference.","display_name":"Def xref","type":"string","property":["getDefXref"],"cardinality":"multi"},"database_xref":{"id":"database_xref","description":"Database cross-reference.","display_name":"DB xref","type":"string","property":["getXref"],"cardinality":"multi"},"isa_partof_closure":{"id":"isa_partof_closure","description":"Ancestral terms (is_a/part_of).","display_name":"Is-a/part-of","type":"string","cardinality":"multi","property":["getRelationIDClosure","BFO:0000050"]},"isa_partof_closure_label":{"id":"isa_partof_closure_label","description":"Ancestral terms (is_a/part_of).","display_name":"Is-a/part-of","type":"string","cardinality":"multi","property":["getRelationLabelClosure","BFO:0000050"],"searchable":true},"regulates_closure":{"id":"regulates_closure","description":"Ancestral terms (regulates, occurs in, capable_of).","display_name":"Ancestor","type":"string","cardinality":"multi","property":["getRelationIDClosure","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"]},"regulates_closure_label":{"id":"regulates_closure_label","description":"Ancestral terms (regulates, occurs in, capable_of).","display_name":"Ancestor","type":"string","cardinality":"multi","property":["getRelationLabelClosure","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"],"searchable":true},"topology_graph_json":{"id":"topology_graph_json","description":"JSON blob form of the local stepwise topology graph. Uses various relations (including regulates, occurs in, capable_of).","display_name":"Topology graph (JSON)","type":"string","property":["getSegmentShuntGraphJSON","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"],"indexed":false},"regulates_transitivity_graph_json":{"id":"regulates_transitivity_graph_json","description":"JSON blob form of the local relation transitivity graph. Uses various relations (including regulates, occurs in, capable_of).","display_name":"Regulates transitivity graph (JSON)","type":"string","property":["getLineageShuntGraphJSON","BFO:0000050","BFO:0000066","RO:0002211","RO:0002212","RO:0002213","RO:0002215","RO:0002216"],"indexed":false},"only_in_taxon":{"id":"only_in_taxon","description":"Only in taxon.","display_name":"Only in taxon","type":"string","property":["getIdentifier"],"searchable":true},"only_in_taxon_label":{"id":"only_in_taxon_label","description":"Only in taxon label.","display_name":"Only in taxon","type":"string","property":["getLabel"],"searchable":true},"only_in_taxon_closure":{"id":"only_in_taxon_closure","description":"Only in taxon closure.","display_name":"Only in taxon (IDs)","type":"string","cardinality":"multi","property":["getRelationLabelClosure","RO:0002160"]},"only_in_taxon_closure_label":{"id":"only_in_taxon_closure_label","description":"Only in taxon label closure.","display_name":"Only in taxon","type":"string","cardinality":"multi","property":["getRelationLabelClosure","RO:0002160"],"searchable":true}}},"variant_disease":{"id":"variant_disease","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 subject_gene^6.3 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Disease","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Disease","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Disease","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Disease","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"variant_gene":{"id":"variant_gene","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Gene","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Gene","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Subject Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Subject Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"variant_genotype":{"id":"variant_genotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 subject_gene^6.3 relation^6.0  object^5.0 object_taxon^4.2 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_closure_label^3.5 object_taxon_closure_label^3.0","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Genotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Genotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Genotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Genotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Genotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Genotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}},"variant_phenotype":{"id":"variant_phenotype","schema_generating":true,"document_category":"generic_association","weight":20,"result_weights":"subject^7.0 subject_taxon^6.5 subject_gene^6.2 relation^6.0  object^5.0 evidence^4.0 source^3.5 is_defined_by^3.0","filter_weights":"subject_taxon_label^3.5","fields":[{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}],"description":"Associations following OBAN model","display_name":"Annotations","boost_weights":"subject^2.0 subject_label^1.0 object^2.0 object_label^1.0","fields_hash":{"subject":{"id":"subject","description":"The CURIE for oban:association_has_subject. May be disease, variant, gene, person, ....","display_name":"Variant","type":"string","property":[]},"subject_label":{"id":"subject_label","description":"Label for association_subject. This will always be rdfs:label. Conventions may vary as to what goes in the label. For genes will be symbol, but we may choose to uniquify by prefixing with species","display_name":"Variant","type":"string","property":[],"searchable":true},"subject_closure":{"id":"subject_closure","description":"Reflexive closure of association_has_subject. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Variant","type":"string","cardinality":"multi","property":[]},"subject_closure_label":{"id":"subject_closure_label","description":"Labels for subject_closure.","display_name":"Variant","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_taxon":{"id":"subject_taxon","description":"Taxonomic class of the subject. This is typically a CURIE of the form NCBITaxon:nnnn.","display_name":"Species","type":"string","property":[]},"subject_taxon_label":{"id":"subject_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"subject_taxon_closure":{"id":"subject_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"subject_taxon_closure_label":{"id":"subject_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_gene":{"id":"subject_gene","description":"Gene related to the subject.","display_name":"Gene","type":"string","property":[]},"subject_gene_label":{"id":"subject_gene_label","description":"Label of the gene.","display_name":"Gene","type":"string","property":[],"searchable":true},"subject_gene_closure":{"id":"subject_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"subject_gene_closure_label":{"id":"subject_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object":{"id":"object","description":"The CURIE for oban:association_has_object. This is often, but not always an ontology class. E.g. for a gene-gene interaction it is an (arbitrary) member of the pair.","display_name":"Phenotype","type":"string","property":[]},"object_label":{"id":"object_label","description":"Label for association_object.","display_name":"Phenotype","type":"string","property":[],"searchable":true},"object_closure":{"id":"object_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Phenotype","type":"string","cardinality":"multi","property":[]},"object_closure_label":{"id":"object_closure_label","description":"Labels for object_class_closure.","display_name":"Phenotype","type":"string","cardinality":"multi","property":[],"searchable":true},"object_taxon":{"id":"object_taxon","description":"Taxonomic class of the object. This is typically a CURIE of the form NCBITaxon:nnnn. This field may be unfilled when used with certain categories","display_name":"Species","type":"string","property":[]},"object_taxon_label":{"id":"object_taxon_label","description":"Label of taxon.","display_name":"Species","type":"string","property":[],"searchable":true},"object_taxon_closure":{"id":"object_taxon_closure","description":"Reflexive closure of taxon. ALWAYS up SubClassOf.","display_name":"Species","type":"string","cardinality":"multi","property":[]},"object_taxon_closure_label":{"id":"object_taxon_closure_label","description":"Labels for taxon_closure.","display_name":"Species","type":"string","cardinality":"multi","property":[],"searchable":true},"evidence_object":{"id":"evidence_object","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_label":{"id":"evidence_object_label","description":"evidence object label","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure":{"id":"evidence_object_closure","description":"Closure for evidence object","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"evidence_object_closure_label":{"id":"evidence_object_closure_label","description":"List of CURIES of all entities that form part of the evidence closure.","display_name":"Evidence","type":"string","cardinality":"multi","property":[]},"id":{"id":"id","description":"A unique identifier (CURIE) for the association. Optional.","display_name":"Association id","type":"uuid","property":[]},"subject_category":{"id":"subject_category","description":"Category of association.subject. Examples: gene, protein, disease, variant","display_name":"Subject category","type":"string","property":[]},"subject_category_label":{"id":"subject_category_label","description":"Label for association_subject_category","display_name":"Subject category","type":"string","property":[]},"subject_description":{"id":"subject_description","description":"A more descriptive label or full name for association_subject. For a gene this may be the full name (as opposed to symbol).","display_name":"Subject description","type":"string","property":[],"searchable":true},"subject_synonym":{"id":"subject_synonym","description":"synonyms for the entity in the association.subject field.","display_name":"Synonym","type":"string","cardinality":"multi","property":[]},"subject_chromosome":{"id":"subject_chromosome","description":"Chromosome related to the subject.","display_name":"Subject Chromosome","type":"string","property":[]},"subject_chromosome_label":{"id":"subject_chromosome_label","description":"Label of the chromosome.","display_name":"Subject Chromosome","type":"string","property":[],"searchable":true},"subject_chromosome_closure":{"id":"subject_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"subject_chromosome_closure_label":{"id":"subject_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"subject_family":{"id":"subject_family","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_family_label":{"id":"subject_family_label","description":"PANTHER families that are associated with this entity.","display_name":"PANTHER family","type":"string","searchable":true,"property":[]},"subject_ortholog_closure":{"id":"subject_ortholog_closure","description":"Reflexive closure of orthology.","display_name":"Ortholog","type":"string","cardinality":"multi","property":[]},"relation":{"id":"relation","description":"A relationship type that connects the subject with object.","display_name":"Relationship","type":"string","property":[]},"relation_label":{"id":"relation_label","description":"Label for association_relation","display_name":"Relationship","type":"string","property":[]},"relation_closure":{"id":"relation_closure","description":"SubPropertyOf reflexive closure for association_relation","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"relation_closure_label":{"id":"relation_closure_label","description":"labels for association_relation_closure","display_name":"Relationship","type":"string","cardinality":"multi","property":[]},"qualifier":{"id":"qualifier","description":"Association qualifier.","display_name":"Qualifier","type":"string","cardinality":"multi","property":[]},"object_category":{"id":"object_category","description":"Category of association.object. Examples: phenotype, function, process, location, tissue, gene. In GO this is called aspect","display_name":"Object type","type":"string","property":[]},"object_category_label":{"id":"object_category_label","description":"Label for association_object_category","display_name":"Object type","type":"string","property":[]},"object_gene":{"id":"object_gene","description":"Gene related to the subject.","display_name":"Object Gene","type":"string","property":[]},"object_gene_label":{"id":"object_gene_label","description":"Label of the gene.","display_name":"Object Gene","type":"string","property":[],"searchable":true},"object_gene_closure":{"id":"object_gene_closure","description":"Reflexive closure of gene.","display_name":"Gene","type":"string","cardinality":"multi","property":[]},"object_gene_closure_label":{"id":"object_gene_closure_label","description":"Labels for gene_closure.","display_name":"Gene","type":"string","cardinality":"multi","property":[],"searchable":true},"object_chromosome":{"id":"object_chromosome","description":"Chromosome related to the subject.","display_name":"Object Chromosome","type":"string","property":[]},"object_chromosome_label":{"id":"object_chromosome_label","description":"Label of the chromosome.","display_name":"Object Chromosome","type":"string","property":[],"searchable":true},"object_chromosome_closure":{"id":"object_chromosome_closure","description":"Reflexive closure of chromosome.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[]},"object_chromosome_closure_label":{"id":"object_chromosome_closure_label","description":"Labels for chromosome_closure.","display_name":"Chromosome","type":"string","cardinality":"multi","property":[],"searchable":true},"object_isoform":{"id":"object_isoform","description":"Assoc object alternate form. E.g. for a canonical protein may be isoforms. Can we generalize this? May be deleted","display_name":"Isoform","type":"string","property":[]},"object_class_closure":{"id":"object_class_closure","description":"Reflexive closure of association_has_object. A list of CURIEs. If an individual, first traverse rdf:tpye. The default closure is subclass, but other may optionally be added depending on what type of association this is and what the class is. E.g. for expression associations, the object is an anatomy class, and the closure will include part_of","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_closure_label":{"id":"object_class_closure_label","description":"Labels for object_class_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_class_secondary_closure":{"id":"object_class_secondary_closure","description":"Optional. This is similar to object_class_closure, but allows for the scenario whereby a different (more inclusive, or exclusive) closure is used. The exact OPs used will depend on the association type. This would ideally be communicating from loader conf through to the client somehow","display_name":"Inferred object","type":"string","cardinality":"multi","property":[]},"object_class_secondary_closure_label":{"id":"object_class_secondary_closure_label","description":"Labels for object_class_secondary_closure.","display_name":"Involved in","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension":{"id":"object_extension","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_label":{"id":"object_extension_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_closure":{"id":"object_extension_closure","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"object_extension_closure_label":{"id":"object_extension_closure_label","description":"Extension class for the annotation.","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[],"searchable":true},"object_extension_json":{"id":"object_extension_json","description":"Extension class for the annotation (JSON).","display_name":"Annotation extension","type":"string","cardinality":"multi","property":[]},"source":{"id":"source","description":"References provided as evidence for a single association in the assertion chain. The assertion may be derived from connecting data from multiple sources, or multiple sources may corroborate the same assertion. References are collected from each source used in this chain when available.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"source_label":{"id":"source_label","description":"Label for source","display_name":"Object","type":"string","cardinality":"multi","property":[],"searchable":true},"source_closure":{"id":"source_closure","description":"CURIEs for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[]},"source_closure_label":{"id":"source_closure_label","description":"Labels for source_closure","display_name":"Object (Inferred)","type":"string","cardinality":"multi","property":[],"searchable":true},"date":{"id":"date","description":"Date of assignment. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable.","display_name":"Date","type":"string","property":[]},"assigned_by":{"id":"assigned_by","description":"Source of association. The association may be derived from multiple source, but this should be the \'outer\' source, if applicable","display_name":"Assigned by","type":"string","property":[]},"is_redundant_for":{"id":"is_redundant_for","description":"Rational for redundancy of annotation.","display_name":"Redundant for","type":"string","property":[]},"evidence_graph":{"id":"evidence_graph","description":"Evidence graph","display_name":"Evidence graph formatted as graphson","type":"string","property":[]},"evidence":{"id":"evidence","description":"Evidence type. In Monarch we may have a chain of assertions that link two entites/terms. This is a list of all evidence types used in that chain.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_label":{"id":"evidence_label","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence label","type":"string","cardinality":"multi","property":[]},"evidence_closure":{"id":"evidence_closure","description":"Closure for evidence_type. Always SubClassof","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_closure_label":{"id":"evidence_closure_label","description":"List of CURIES of all entities that form part of the evidence graph.","display_name":"Evidence type","type":"string","cardinality":"multi","property":[]},"evidence_main_object":{"id":"evidence_main_object","description":"A list of CURIEs for the primary object that supports an association. This will be a subset of evidence_object.","display_name":"Evidence from (main)","type":"string","cardinality":"multi","property":[]},"evidence_main_object_label":{"id":"evidence_main_object_label","description":"evidence main object label","display_name":"Evidence main label","type":"string","cardinality":"multi","property":[]},"citation":{"id":"citation","description":"citation.","display_name":"Reference","type":"string","cardinality":"multi","property":[]},"is_defined_by":{"id":"is_defined_by","description":"Associations are obtained from the source(s) listed. More than one source indicates the association may be derived from connecting data from multiple sources, or multiple sources have corroborated the same assertion.","display_name":"Source","type":"string","cardinality":"multi","property":[]}}}}';

    var golr_json = JSON.parse(golr_json_string);

    var gconf = new bbop.golr.conf(golr_json);
    var golr_loc = 'https://solr.monarchinitiative.org/solr/ontology/';

    var golr_manager_for_disease = new bbop.golr.manager.jquery(golr_loc, gconf);
    golr_manager_for_disease.set_results_count(50);

    // dirty trick to make jQuery avaiable in the golrmanager's scope.
    golr_manager_for_disease.JQ = jQuery;
    that.disease_autocomplete_options = function(onChangeFunc) {
      return {
        onChange: onChangeFunc,
        required: true,
        optionDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            '</div>';
        },
        itemDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            '</div>';
        },
        valueField: 'id',
        searchField: ['id', 'annotation_class_label_searchable'],
        filterData: function() {
          var filter = [['isa_partof_closure', rootDiseaseId]];
          return filter;
        },
        queryData: function(query) {
          var result = 'annotation_class_label_searchable:*' + query + '*';
          if (that.modelType === 'hpo') {
            result += ' AND id:OMIM\\:*';
          }
          return result;
        },
        golrManager: golr_manager_for_disease
      };
    };

    var golr_manager_for_phenotype = new bbop.golr.manager.jquery(golr_loc, gconf);
    golr_manager_for_phenotype.set_results_count(10);
    // dirty trick to make jQuery avaiable in the golrmanager's scope.
    golr_manager_for_phenotype.JQ = jQuery;
    that.phenotype_autocomplete_options = function(onChangeFunc) {
      return {
        onChange: onChangeFunc,
        required: true,
        optionDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            '</div>';
        },
        itemDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            '</div>';
        },
        valueField: 'id',
        searchField: ['id', 'annotation_class_label_searchable'],
        filterData: function() {
          var filter = [['isa_partof_closure', rootPhenotypeId]];
          return filter;
        },
        queryData: function(query) {
          var result = 'annotation_class_label_searchable:*' + query + '*';
          if (that.modelType === 'hpo') {
            result += ' AND id:HP\\:*';
          }
          return result;
        },
        golrManager: golr_manager_for_phenotype
      };
    };


    var golr_manager_for_ageofonset = new bbop.golr.manager.jquery(golr_loc, gconf);
    golr_manager_for_ageofonset.set_results_count(50);
    // dirty trick to make jQuery avaiable in the golrmanager's scope.
    golr_manager_for_ageofonset.JQ = jQuery;
    that.ageofonset_autocomplete_options = function(onChangeFunc) {
      return {
        onChange: onChangeFunc,
        optionDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            '</div>';
        },
        itemDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            '</div>';
        },
        valueField: 'id',
        searchField: ['id', 'annotation_class_label_searchable'],
        filterData: function() {
          return [['isa_partof_closure', rootOnsetId]];
        },
        queryData: function(query) {
          // return 'isa_partof_closure_label_searchable:Onset AND id:*' + query.replace(':', '\\:').toUpperCase() + '*';
          return 'annotation_class_label_searchable:*' + query + '*';
        },
        golrManager: golr_manager_for_ageofonset
      };
    };


    var golr_manager_for_evidence = new bbop.golr.manager.jquery(golr_loc, gconf);
    golr_manager_for_evidence.set_results_count(50);
    that.evidence_autocomplete_options = function(onChangeFunc) {
      return {
        onChange: onChangeFunc,
        optionDisplay: function(item, escape) {
          return '<div>' +
            item.id + ' (' + item.annotation_class_label_searchable + ')' +
            // item.id + ' (' + item.synonym_searchable + ')' +
            '</div>';
        },
        itemDisplay: function(item, escape) {
          var result =
              '<div>' +
                item.id +
                ' (' + item.annotation_class_label_searchable + ')' +
              '</div>';
          return result;
        },
        valueField: 'id',
        // searchField: ['id', 'annotation_class_label_searchable'],
        searchField: ['id', 'synonym_searchable'],

        filterData: function() {
          var filter = [['isa_partof_closure', rootEvidenceId]];
          return filter;
        },
        queryData: function(query) {
          var result = 'synonym_searchable:*' + query + '*';
          return result;
        },
        golrManager: golr_manager_for_evidence
      };
    };
  }

  this.initializeSubjectAutocomplete = function() {
    var subject_selector = '#select_subject_default';
    var subject_selectize = jQuery(subject_selector)[0];

    if (subject_selectize && subject_selectize.selectize) {
      subject_selectize = subject_selectize.selectize;
      // console.log('initializeSubjectAutocomplete FOUND', subject_selector, subject_selectize);
    }
    else {
      Solrautocomplete.createSolrAutocompleteForElement(
        '#select_subject_default', that.disease_autocomplete_options(function(value) {
        var title = subject_selectize.options[value].annotation_class_label;
        $timeout(function() {
          value = value.replace(/^MESH:/, 'MeSH:');
          that.newSubject = value;
          that.newTitle = title;
        }, 10);
      }));
      subject_selectize = jQuery(subject_selector)[0].selectize;
      // console.log('initializeSubjectAutocomplete CREATE', subject_selector, subject_selectize);
    }

    $timeout(function() {
      subject_selectize.focus();
    }, 300);
  };

  this.initializeRowAutocomplete = function(rowIndex) {
    // console.log('initializeRowAutocomplete', rowIndex);
    this.editingPhenotype = false;
    var phenotype_selector = '#select_phenotype_' + rowIndex;
    var phenotype_selectize = jQuery(phenotype_selector)[0];

    if (phenotype_selectize && phenotype_selectize.selectize) {
      phenotype_selectize = phenotype_selectize.selectize;
      // console.log('initializeRowAutocomplete FOUND', phenotype_selector, phenotype_selectize);
    }
    else {
      Solrautocomplete.createSolrAutocompleteForElement(phenotype_selector, that.phenotype_autocomplete_options(function(value) {
        that.selected_phenotype = value;
        if (phenotype_selectize.options[value]) {
          var title = phenotype_selectize.options[value].annotation_class_label;
          that.selected_phenotype_label = value;
          if (that.editingPhenotype) {
            that.$timeout(function() {
              that.$scope.$broadcast('uiGridEventEndCellEdit');
            }, 100);
          }
        }
      }));
      phenotype_selectize = jQuery(phenotype_selector)[0].selectize;
      // console.log('initializeRowAutocomplete CREATE', phenotype_selector, phenotype_selectize);

      var ageofonset_selector = '#select_ageofonset_' + rowIndex;
      Solrautocomplete.createSolrAutocompleteForElement(ageofonset_selector, that.ageofonset_autocomplete_options(function(value) {
        $timeout(function() {
          that.selected_ageofonset = value;
        }, 10);
      }));
    }
  };


  this.cleanupRowAutocomplete = function(rowIndex) {
  };

  this.init();
}




var NoctuaBasicControllerBundle = [
  '$q', '$scope', '$animate', '$timeout', '$interval', '$location', '$anchorScroll',
  'toastr', '$window', '$rootScope', 'uiGridConstants',
  NoctuaBasicController];

var app = angular.module('noctuaBasicApp');
app.controller('NoctuaBasicController', NoctuaBasicControllerBundle);



if (USE_UI_GRID) {
  app.filter('phenotypeLabel', function () {
    return function(phenotype) {
      return phenotype ? phenotype[1] : '';
    };
  });

  /* eslint no-inner-declarations: 0 */
  function uiSelectWrapIcon($document, uiGridEditConstants) {
    return function link($scope, $elm, $attr) {
      function docClick(evt) {
        if (jQuery(evt.target).closest('.ui-select-container').size() === 0) {
          $scope.$emit(uiGridEditConstants.events.END_CELL_EDIT);
          $document.off('click', docClick);
        }
      }
      $document.on('clickIcon', docClick);
    };
  }
  uiSelectWrapIcon.$inject = ['$document', 'uiGridEditConstants'];


  function uiSelectWrap($document, uiGridEditConstants) {
    function link($scope, $elm, $attr, uiGridCtrl) {
      function hideOnCellnav() {
        if (uiGridCtrl && uiGridCtrl.grid.api.cellNav) {
          var dereg = uiGridCtrl.grid.api.cellNav.on.navigate($scope, function (newRowCol, oldRowCol) {
            if ($scope.col.colDef.enableCellEditOnFocus) {
              if (newRowCol.row !== $scope.row || newRowCol.col !== $scope.col) {
                $scope.$emit(uiGridEditConstants.events.END_CELL_EDIT);
                dereg();
              }
            }
          });
        }
      }

      function docClick(evt) {
        if (jQuery(evt.target).closest('.ui-select-container').size() === 0) {
          $scope.$emit(uiGridEditConstants.events.END_CELL_EDIT);
          $document.off('click', docClick);
        }
      }

      $document.on('click', docClick);

      // set focus at start of edit
      $scope.$on(uiGridEditConstants.events.BEGIN_CELL_EDIT, function (evt, args) {
        hideOnCellnav();
      });
    }

    return {
      require: '?^uiGrid',
      link: link
    };
  }
  uiSelectWrap.$inject = ['$document', 'uiGridEditConstants'];

  app.directive('uiSelectWrap', uiSelectWrap);
  app.directive('uiSelectWrapIcon', uiSelectWrapIcon);

  // http://stackoverflow.com/a/32717235
  app.config(
    ['$provide',
    function($provide) {
      $provide.decorator('uiGridCellNavService',
        ['$delegate', 'uiGridConstants', 'uiGridCellNavConstants',
        function ($delegate, uiGridConstants, uiGridCellNavConstants) {
          var getDirection = $delegate.getDirection;
          $delegate.getDirection = function (evt) {
            if (evt.keyCode === uiGridConstants.keymap.LEFT ||
              (evt.keyCode === uiGridConstants.keymap.TAB && evt.shiftKey)) {
              return uiGridCellNavConstants.direction.LEFT;
            }
            if (evt.keyCode === uiGridConstants.keymap.RIGHT ||
              evt.keyCode === uiGridConstants.keymap.TAB) {
              return uiGridCellNavConstants.direction.RIGHT;
            }

            if (evt.keyCode === uiGridConstants.keymap.UP ||
              (evt.keyCode === uiGridConstants.keymap.ENTER && evt.shiftKey) ) {
              return uiGridCellNavConstants.direction.UP;
            }

            if (evt.keyCode === uiGridConstants.keymap.PG_UP) {
              return uiGridCellNavConstants.direction.PG_UP;
            }
            // override listener for key ENTER
            if (evt.keyCode === uiGridConstants.keymap.ENTER) {
              return uiGridCellNavConstants.direction.RIGHT;
            }
            // delete listener for key ENTER
            if (evt.keyCode === uiGridConstants.keymap.DOWN) {
              return uiGridCellNavConstants.direction.DOWN;
            }
            // original code
            //    if (evt.keyCode === uiGridConstants.keymap.DOWN ||
            //      evt.keyCode === uiGridConstants.keymap.ENTER) {
            //      return uiGridCellNavConstants.direction.DOWN;
            //    }

            if (evt.keyCode === uiGridConstants.keymap.PG_DOWN) {
              return uiGridCellNavConstants.direction.PG_DOWN;
            }

            return null;
          };
          return $delegate;
        }]);
    }]);
}
